### ISSUE TITLE
Document that Cholesky only uses the lower triangular part

### ISSUE THREAD

**Issue by vks** [AtheMathmo/rulinalg#188]
This informs the user that it is possible to only initialize the lower triangular part, saving a few operations.

**Comment by Andlon** [comment]
Yes, this makes sense - on the other hand, if we provide that guarantee we cannot easily change our minds later, because changing this behavior would subtly break user's code without warning (since there are no API breakages or similar).

I am planning to rewrite the Cholesky decomposition to use BLAS-3 in the future, and I can't say for sure if we'd still only access the lower triangular part then - but I think so. I suggest we keep this issue open and revisit when we are more confident in providing such a guarantee.
